crypto/bio/bio_lib.c

```diff
@@ -12,23 +12,63 @@
 #include <openssl/crypto.h>
 #include "bio_lcl.h"
 #include "internal/cryptlib.h"
-#include <openssl/stack.h>
+
+static BIO nilb = {
+    NULL, NULL, NULL,   /* method, callback, cb_arg */
+    0, 1, 0,            /* init, shutdown, flags */
+    0, 0,               /* retry_reason, num */
+    NULL, NULL, NULL,   /* ptr, next_bio, prev_bio */
+    1,                  /* references */
+    0L, 0L,             /* num_read, num_write */
+    { NULL },           /* ex_data */
+    NULL                /* lock */
+};
 
 BIO *BIO_new(const BIO_METHOD *method)
 {
-    BIO *ret = OPENSSL_malloc(sizeof(*ret));
+    BIO *bio = OPENSSL_malloc(sizeof(*bio));
```

Comment:
as with most other structures (you did much of it :) just use zalloc directly.  don't need nilb.